### PR TITLE
Generic: Handle incorrect json

### PR DESCRIPTION
**Pull request by kiblik**
Refuse to process incorrect generic JSON - if some keys are missing or if it is not allowed to modify them

### CODE REVIEW

File: dojo/tools/generic/parser.py [hunk 1]
```diff
@@ -52,14 +53,30 @@ def _get_test_json(self, data):
...
             # check for required keys
             required = ['title', 'date', 'severity', 'description']
             missing = []
             for field in required:
-                if field not in data:
+                if field not in item.keys():
                     missing.append(field)
             if missing:
-                raise Exception(f"Required fields are missing: {missing}")
+                raise ValidationError(f"Required fields are missing: {missing}")
```
Comment (damiencarol) [inline dojo/tools/generic/parser.py:79]: Please use `ValueError` instead. Using a Django error could be misleading.
```suggestion
                raise ValueError(f"Required fields are missing: {missing}")
```

**Review by damiencarol — CHANGES_REQUESTED** [review]

File: dojo/tools/generic/parser.py (hunk 1 above)
Comment (kiblik) [reply dojo/tools/generic/parser.py:79]: I see your point. But if we will use only `ValueError`, the user will receive `HTTP-500` with 
```json
{
  "message": "Internal server error, check logs for details"
}
```
and he/she/they will have to contact the admin for logs to identify small mistakes.
In the case of `ValidationError`, the user will be able to see the error message in the response and identify the problem and solve it.

What about using `ValueError`s in parsers and "rewrap" them to `ValidationError` in (re)importer?

**Review by Maffooch — APPROVED** [review]

**Review by devGregA — APPROVED** [review]

**Review by blakeaowens — APPROVED** [review]
LGTM

**Review by mtesauro — APPROVED** [review]
Approved

**Comment by mtesauro** [comment]
@damiencarol It looks like you're the last person with proposed changes to this.

If it's good by you, feel free to merge.

**Review by damiencarol — APPROVED** [review]

**Comment by damiencarol** [comment]
@kiblik the final version is good!

**Pull request merged**